src/producer.rs

```diff
@@ -119,28 +120,49 @@ fn zip_producer(tmp_dir: &Path, zip_files: &[&String], queue: &WorkQueue, ignore
                 let stem = path.file_stem().unwrap().to_str().unwrap();
 
                 let physical_gcno_path = path.with_file_name(format!("{}_{}.gcno", stem, 1));
-                extract_file(&mut gcno_file, &physical_gcno_path);
+                let mut gcno_buf: Option<Arc<Vec<u8>>> = None;
+                if is_llvm {
+                    let mut buffer: Vec<u8> = Vec::new();
+                    gcno_file.read_to_end(&mut buffer).expect("Failed to read gcno file");
+                    gcno_buf = Some(Arc::new(buffer));
+                } else {
+                    extract_file(&mut gcno_file, &physical_gcno_path);
+                }
 
                 let gcda_path_in_zip = gcno_path_in_zip.with_extension("gcda");
 
                 for (num, &mut (gcda_archive_name, ref mut gcda_archive)) in gcda_archives.iter_mut().enumerate() {
                     let gcno_path = path.with_file_name(format!("{}_{}.gcno", stem, num + 1));
-
                     if let Ok(mut gcda_file) = gcda_archive.by_name(&gcda_path_in_zip.to_str().unwrap().replace("\\", "/")) {
-                        // Create symlinks.
-                        if num != 0 {
-                            fs::hard_link(&physical_gcno_path, &gcno_path).expect(format!("Failed to create hardlink {}", gcno_path.display()).as_str());
+                        if is_llvm {
+                            let mut gcda_buf: Vec<u8> = Vec::new();
+                            gcda_file.read_to_end(&mut gcda_buf).expect("Failed to read gcda file");
+                            let gcno_stem = path.with_file_name(format!("{}_{}", stem, num + 1));
+                            let gcno_stem = gcno_stem.to_str().expect("Failed to create stem file string");
+
+                            queue.push(Some(WorkItem {
+                                format: ItemFormat::GCNO,
+                                item: ItemType::Buffers(GcnoBuffers {stem: gcno_stem.to_string(),
+                                                                     gcno_buf: Arc::clone(&gcno_buf.clone().unwrap()),
```

Comment:
Let's try to use match to avoid cloning, as cloning the Option probably clones the buffer too: https://doc.rust-lang.org/std/option/enum.Option.html#method.cloned.